File: problems/problems_Interview_05__01/Solution.java
package problems.problems_Interview_05__01;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public int insertBits(int N, int M, int i, int j) {
        
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        int N = Integer.parseInt(inputJsonValues[0]);
		int M = Integer.parseInt(inputJsonValues[1]);
		int i = Integer.parseInt(inputJsonValues[2]);
		int j = Integer.parseInt(inputJsonValues[3]);
        return JSON.toJSON(insertBits(N, M, i, j));
    }
}

File: problems/problems_Interview_05__01/solution.go
package problemInterview_05__01

import (
	"encoding/json"
	"log"
	"strings"
)

func insertBits(N int, M int, i int, j int) int {
    
}

func Solve(inputJsonValues string) any {
	inputValues := strings.Split(inputJsonValues, "\n")
	var N int
	var M int
	var i int
	var j int

	if err := json.Unmarshal([]byte(inputValues[0]), &N); err != nil {
		log.Fatal(err)
	}
	if err := json.Unmarshal([]byte(inputValues[1]), &M); err != nil {
		log.Fatal(err)
	}
	if err := json.Unmarshal([]byte(inputValues[2]), &i); err != nil {
		log.Fatal(err)
	}
	if err := json.Unmarshal([]byte(inputValues[3]), &j); err != nil {
		log.Fatal(err)
	}

	return insertBits(N, M, i, j)
}
